rspotify-http/src/ureq.rs

```diff
@@ -41,21 +51,28 @@ impl UreqClient {
         }
 
         log::info!("Making request {:?}", request);
+        // Converting errors from ureq into our custom error types
         match send_request(request) {
-            // Successful request
             Ok(response) => response.into_string().map_err(Into::into),
-            // HTTP status error
-            Err(ureq::Error::Status(_, response)) => Err(HttpError::from_response(response)),
-            // Some kind of IO/transport error
-            Err(err) => Err(HttpError::Request(err.to_string())),
+            Err(err) => match err {
+                ureq::Error::Status(_, response) => Err(UreqError::StatusCode(response)),
+                ureq::Error::Transport(transport) => Err(UreqError::Transport(transport)),
+            },
         }
     }
 }
 
 #[sync_impl]
 impl BaseHttpClient for UreqClient {
+    type Error = UreqError;
+
     #[inline]
-    fn get(&self, url: &str, headers: Option<&Headers>, payload: &Query) -> HttpResult<String> {
+    fn get(
+        &self,
+        url: &str,
+        headers: Option<&Headers>,
+        payload: &Query,
+    ) -> Result<String, UreqError> {
```

Comment:
You could simply return `-> Result<String, Self::Error>` as what `reqwest` did.

Reply:
Yup you're right.